### ISSUE TITLE
"Nan" appears after data processing

### ISSUE THREAD

**Issue by Sylva-Lin** [Pointcept/PointTransformerV3#16]
![屏幕截图 2024-03-12 215748](https://github.com/Pointcept/PointTransformerV3/assets/105730616/7d127cc0-d8b6-40c2-90d8-72107ae189c1)
Problem description: In the training stage, "nan" appears after the coding layer processes the input features, and after debugging, it is located in the position of the "Block" function, where the data starts to appear "nan". This error was initially discovered because the loss in the training phase was "nan", which caused it not to run.
Dataset and Config: S3DIS, batch_size = 4
![屏幕截图 2024-03-12 220746](https://github.com/Pointcept/PointTransformerV3/assets/105730616/c4f57cf6-6246-491d-9b24-1f14a7c3f779)
I don't know what caused this, can you give me some advice?

**Comment by Gofinge** [comment]
I think It might be the common issue we describe as "gradient explosion", and a change in batch size (12->4) should cause it. Typically, I recommend keeping batch size unchanged as we only ensure the default setting can reproduce our performance.  Yet if you do need to reduce batch size, please change the learning rate accordingly, a smaller batch size with a smaller learning rate. Also, a strategy to make large batch size applicable to small GPU is gradient accumulation. I have not added it to our Pointcept, you can try it by yourself.

**Comment by Sylva-Lin** [comment]
> I think It might be the common issue we describe as "gradient explosion", and a change in batch size (12->4) should cause it. Typically, I recommend keeping batch size unchanged as we only ensure the default setting can reproduce our performance. Yet if you do need to reduce batch size, please change the learning rate accordingly, a smaller batch size with a smaller learning rate. Also, a strategy to make large batch size applicable to small GPU is gradient accumulation. I have not added it to our Pointcept, you can try it by yourself.

I looked at the learning rate at the time of my training and it was 0.006 on the 8th epoch, is this trend anomalous?
![屏幕截图 2024-03-12 225441](https://github.com/Pointcept/PointTransformerV3/assets/105730616/00625b40-8830-4382-b157-5b04894113b2)

**Comment by Sylva-Lin** [comment]
I'm using the default learning rate configuration, should I set 0.06 to a smaller value?
![屏幕截图 2024-03-12 225945](https://github.com/Pointcept/PointTransformerV3/assets/105730616/61952344-c7e4-4a66-8b12-b93ad432a795)

**Comment by Sylva-Lin** [comment]
> Try 0.002?
Hi, I reduced the learning rate to 0.002 and now I can train normally, but the accuracy of the validation set has been poor, it seems to be an overfit?
![屏幕截图 2024-03-13 194438](https://github.com/Pointcept/PointTransformerV3/assets/105730616/a0692fd2-68ae-4403-86f2-70dcb6dd3d94)

**Comment by Gofinge** [comment]
The batch size itself is a significant parameter that affects final performance. Consequently, as in my previous answer (https://github.com/Pointcept/PointTransformerV3/issues/16#issuecomment-1991803702), I won't recommend adjusting the batch size. You can consider the strategy I mentioned: gradient accumulation.

**Comment by Sylva-Lin** [comment]
![屏幕截图 2024-03-15 125123](https://github.com/Pointcept/PointTransformerV3/assets/105730616/e63ac611-09f9-48e6-ac4a-1a8908a334cf)
Hi, I tested after training according to this setting, although the results are not as good as the results of the paper, they are also much better than the results of the evaluation, in order to avoid OOM during the test, I set the grid_size to 0.03, and the crop was introduced during the evaluation, is it possible that the results of the evaluation phase are unstable due to this reason?

**Comment by Gofinge** [comment]
The performance is too low to blame it on instability. As mentioned, batch size significantly influences performance. Try our default batch size and increase the base grid size to 0.04. also don't forget to adjust the model parameters.
